lib/ansible/galaxy/dependency_resolution/dataclasses.py

```diff
@@ -418,6 +455,30 @@ def from_requirement_dict(cls, collection_req, art_mgr, validate_signature_optio
                 'index server.',
             )
 
+        if req_type != 'git' and req_version != '*':
+            if not HAS_PACKAGING:
+                raise AnsibleError("Failed to import packaging, check that a supported version is installed")
+
+            try:
+                specifier = next(iter(
+                    SpecifierSet(
```

Comment:
I don't think that such version conversions belong in this module. There was another for such utils somewhere.